### ISSUE TITLE
'String' and 'String + Format' yields different results for empty inputs

### ISSUE THREAD

**Issue by igorsantos07** [moment/moment#2779]
Not sure if this is a desired behaviour, but it seems like a bug for me.
```
moment('').format() // Invalid date, ok
moment('').format('X') // Invalid date, ok
moment('').unix() // NaN, ok
moment('', 'DD/MM/YYYY').format() // Invalid date, ok
moment('', 'DD/MM/YYYY').format('X') // Invalid date, ok
moment('', 'DD/MM/YYYY').unix() // 1448668800, wat
```
1448668800 apparently is the timestamp for today's midnight. But shouldn't both yield exactly the same results? Shouldn't `format('X')` and `unix()` return the same?

**Comment by alburthoffman** [comment]
the unix function is impleted like this:

    function unix () {
        return Math.floor(+this / 1000);
    }

And I'm very surprise that +this convert itself to the unix timestamp.

**Comment by mj1856** [comment]
This isn't so much a bug with the `unix` function, but rather that [parsing with a format](https://github.com/moment/moment/blob/master/src/lib/create/from-string-and-format.js) always produces a value in `_a`, regardless of validity.

```javascript
moment('', ' ')._a   // [2015, 11, 3, 0, 0, 0, 0]
moment('')._a   // undefined
```

**Comment by mj1856** [comment]
Note that it occurs regardless of strict mode setting.

**Comment by ichernev** [comment]
This was fixed in #2624
